app/services/iiif_metadata_service.rb

```diff
@@ -53,5 +53,7 @@ def json
                   data['@id'] = @canonical_url
                 end
               end
+  rescue JSON::ParserError
+    raise "There was a problem, #{@url} returned invalid JSON."
```

Comment:
Can you provide the response in the message too? Also, how about making a custom error?  How about `Stacks::UnexpectedMetadataResponse`?

Reply:
For sure.